modules/axdriver/src/bus/pci.rs

```diff
@@ -84,6 +88,37 @@ fn config_pci_device(
 
 impl AllDevices {
     pub(crate) fn probe_bus_devices(&mut self) {
+        self.probe_pci_devices();
+        self.probe_pcie_devices();
+    }
+
+    fn probe_pcie_devices(&mut self) {
+        let base_vaddr = phys_to_virt(axconfig::PCI_ECAM_BASE.into());
+        let mut root = RootComplexGeneric::new(NonNull::new(base_vaddr.as_mut_ptr()).unwrap());
+
+        let headers = root.enumerate_no_modify(None).collect::<Vec<_>>();
```

Comment:
if you can convert it into slice.

deps related with `alloc` can be removed

Reply:
Need mutable borrow `root` later, so need two step.